### PR TITLE
Added a new command !wfr-upload-file to upload a file with a specified file name.

### PR DESCRIPTION
**Pull request by spearmin10**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
N/A

## Description
Added a new command `!wfr-upload-file` to upload a file with a specified file name.

## Screenshots
N/A

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/WebFileRepository/ReleaseNotes/1_0_7.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Web File Repository
+
+- Added a new command: !wfr-upload-file
```
Comment (GuyAfik) [inline Packs/WebFileRepository/ReleaseNotes/1_0_7.md:6]:
```suggestion
- Added the command **wfr-upload-file** which allows to upload files to the repository.
```

File: Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py [hunk 2]
```diff
@@ -4107,6 +4107,33 @@ def command_reset(args: Dict[str, str], settings: Settings) -> str:
...
+
+    data = {
+        'q': 'upload',
+        'dir': args.get('upload_directory', '/'),
+        'extract': 'true' if argToBoolean(args.get('extract_archive', 'false')) else 'false',
+    }
+    resp = client._http_request('POST', data=data, files=files, raise_on_status=True)
+    if not resp.get('success'):
+        raise ValueError(f'Failed to upload a file: {resp.get("message")}')
+    return 'Done.'
```
Comment (GuyAfik) [inline Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py:4134]:
```suggestion
    return CommandResults(readable_output=f'uploaded file {name} successfully.')
```

please update README after implementing this.

**Review by GuyAfik — COMMENTED** [review]
@spearmin10 Thank you for your contribution! nicely done!

see my comments and let me know if you have any questions, you can reach me in DFIR slack.

File: Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py [hunk 3]
```diff
@@ -4107,6 +4107,33 @@ def command_reset(args: Dict[str, str], settings: Settings) -> str:
...
+
+    :param args: The parameters which were given to the command.
+    :param settings: The instance settings.
+    """
+    client = new_client(detect_service_ip_port(settings), settings)
+
+    res = demisto.getFilePath(args.get('entry_id'))
+    path = res['path']
+    if not (name := args.get('file_name')):
+        name = res['name']
```
[the remark this answers is not in the document]
Comment (spearmin10) [reply Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py]: @GuyAfik OK, I've fixed it.

File: Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py [hunk 4]
```diff
@@ -4107,6 +4107,33 @@ def command_reset(args: Dict[str, str], settings: Settings) -> str:
...
+
+    with open(path, 'rb') as f:
+        files = [('file', [name, f.read()])]
+
+    data = {
+        'q': 'upload',
+        'dir': args.get('upload_directory', '/'),
+        'extract': 'true' if argToBoolean(args.get('extract_archive', 'false')) else 'false',
```
[the remark this answers is not in the document]
Comment (spearmin10) [reply Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py:4130]: @GuyAfik `extract` needs to take "true" or "false" in string, not boolean. So the original code is right.

File: Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py (hunk 2 above)
Comment (spearmin10) [reply Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py:4135]: @GuyAfik All the commands returns 'Done' (e.g. !command_upload_files) when it has successfully done. Also it checks the test scripts. That's a manner of this integration. I think it should align with other commands.

File: Packs/WebFileRepository/ReleaseNotes/1_0_7.md (hunk 1 above)
Comment (spearmin10) [reply Packs/WebFileRepository/ReleaseNotes/1_0_7.md]: OK, I've fixed it.

File: Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py:4130]: but `args.get('extract_archive', 'false')` would give you it back as a string as far as i know.
do you think it returns a boolean?

File: Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py (hunk 2 above)
Comment (GuyAfik) [reply Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py:4135]: agreed, lets keep your implementation

File: Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.py:4130]: you have predefined values "false and true", you can setup a default value of "false" and then you won't need this additional code.

however, i don't see any harm in keeping it this way.

**Comment by GuyAfik** [comment]
@spearmin10 let me know when its ready for re-review

**Comment by spearmin10** [comment]
@GuyAfik Please schedule a meeting on my free time of my calender if need to discuss for your review.

**Comment by ShirleyDenkberg** [comment]
@GuyAfik Doc review completed.

File: Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.yml [hunk 5]
```diff
@@ -115,6 +115,24 @@ script:
...
+    - description: The name of the file
+      name: file_name
+    - auto: PREDEFINED
+      defaultValue: "false"
+      description: Set to true to extract files to archive files, otherwise false.
+      name: extract_archive
+      predefined:
+      - "true"
+      - "false"
+    - defaultValue: /
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.yml:131]: @spearmin10 did you can the chance to take a look at it?

[the remark this answers is not in the document]
Comment (spearmin10) [reply Packs/WebFileRepository/Integrations/WebFileRepository/WebFileRepository.yml:131]: That was generated from exporting from XSOAR, and formated by `demisto-sdk format `, was not modified by manual. so it shouldn't be a problem.

**Comment by GuyAfik** [comment]
@spearmin10 looks great, wonderful work! do you mind that sending a picture in the war-room showing its working?
then ill merge it.

**Review by GuyAfik — APPROVED** [review]

**Comment by spearmin10** [comment]
> @spearmin10 looks great, wonderful work! do you mind that sending a picture from the war-room showing its working? then ill merge it.
OK. This is a screen shot that I've run the comment.
image

**Pull request merged**